### PR TITLE
[AssetMapper] Better public without digest

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | None
| License       | MIT
| Doc PR        | Not needed

We have a `MappedAsset::getPublicPathWithoutDigest()` method, which would return something like `/assets/file2.js`. This, sort of odd method, is super useful, for example, when generating the importmap, where we want to communicate to the browser that when it searches for `/assets/file2.js` (e.g. because inside `assets/file1.js` there is an `import './file1.js'`), here it should use the digested path - e.g.

```
'/assets/file2.js' => '/assets/file2.1234abcd.js',
```

Anyway, the way that I calculated this originally was a little sketchy. In truth, we have this info very easily in `AssetMapper`, so now we pass it in. This also allows an "asset compiler" to figure out which "directory" the final asset will be in, which isn't possible with "public path" as it's still being generated. That little nice detail will be needed in StimulusBundle :).

Cheers!

### CODE REVIEW

File: src/Symfony/Component/AssetMapper/AssetMapper.php [hunk 1]
```diff
@@ -239,11 +245,7 @@ private function getMimeType(string $logicalPath): ?string
 
         $extension = pathinfo($logicalPath, \PATHINFO_EXTENSION);
 
-        if (!isset($this->extensionsMap[$extension])) {
-            throw new \LogicException(sprintf('The file extension "%s" from "%s" does not correspond to any known types in the asset mapper. To support this extension, configure framework.asset_mapper.extensions.', $extension, $logicalPath));
-        }
-
-        return $this->extensionsMap[$extension];
+        return $this->extensionsMap[$extension] ?? null;
```
Comment (weaverryan) [inline src/Symfony/Component/AssetMapper/AssetMapper.php:248]: This is unrelated. I had a file that mentioned a sourcemap - so `foo.js.map` - and the system exploded because we didn't know the mime type of `.map`. The mime type is only important for (A) the dev server to return that asset with the correct`Content-Type` and (B) for asset compilers, if they rely on the mime type. So I think this is a reasonable thing to do.

**Review by stof — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @weaverryan.
